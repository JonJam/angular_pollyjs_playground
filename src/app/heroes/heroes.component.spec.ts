import { createRoutingFactory } from '@ngneat/spectator/jest';
import {HeroesComponent} from './heroes.component';
import {AppModule} from '../app.module';

describe('HeroesComponent', () => {
  const createComponent = createRoutingFactory({
    component: HeroesComponent,
    imports: [AppModule],
    stubsEnabled: false,
    detectChanges: false,
  });

  // TODO Exact magic here for detectChanged vs whenStable
  // NOTE: This will make real API requests now
  it('renders something', async () => {
    const spectator = createComponent();

    // Trigger ngOnInit
    spectator.detectChanges();

    // Await for promises to resolve
    await spectator.fixture.whenStable();

    // Trigger ngOnInit
    spectator.detectChanges();

    // TODO Change to not be snapshot
    expect(spectator.fixture).toMatchSnapshot();
  });
});