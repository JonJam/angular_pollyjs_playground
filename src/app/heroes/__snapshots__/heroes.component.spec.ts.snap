// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroesComponent renders something 1`] = `
<app-heroes
  __ngContext__="0"
  heroService={[Function HeroService]}
  heroes={[Function Array]}
>
  <h2>
    My Heroes
  </h2><div>
    <label
      for="new-hero"
    >
      Hero name: 
    </label>
    <input
      id="new-hero"
    />
    <button
      class="add-button"
      type="button"
    >
       Add hero 
    </button>
  </div><ul
    class="heroes"
  >
    <li>
      <a
        href="/detail/1"
      >
        <span
          class="badge"
        >
          1
        </span>
         Leanne Graham 
      </a>
      <button
        class="delete"
        title="delete hero"
        type="button"
      >
        x
      </button>
    </li>
    <li>
      <a
        href="/detail/2"
      >
        <span
          class="badge"
        >
          2
        </span>
         Ervin Howell 
      </a>
      <button
        class="delete"
        title="delete hero"
        type="button"
      >
        x
      </button>
    </li>
    <li>
      <a
        href="/detail/3"
      >
        <span
          class="badge"
        >
          3
        </span>
         Clementine Bauch 
      </a>
      <button
        class="delete"
        title="delete hero"
        type="button"
      >
        x
      </button>
    </li>
    <li>
      <a
        href="/detail/4"
      >
        <span
          class="badge"
        >
          4
        </span>
         Patricia Lebsack 
      </a>
      <button
        class="delete"
        title="delete hero"
        type="button"
      >
        x
      </button>
    </li>
    <li>
      <a
        href="/detail/5"
      >
        <span
          class="badge"
        >
          5
        </span>
         Chelsey Dietrich 
      </a>
      <button
        class="delete"
        title="delete hero"
        type="button"
      >
        x
      </button>
    </li>
    <li>
      <a
        href="/detail/6"
      >
        <span
          class="badge"
        >
          6
        </span>
         Mrs. Dennis Schulist 
      </a>
      <button
        class="delete"
        title="delete hero"
        type="button"
      >
        x
      </button>
    </li>
    <li>
      <a
        href="/detail/7"
      >
        <span
          class="badge"
        >
          7
        </span>
         Kurtis Weissnat 
      </a>
      <button
        class="delete"
        title="delete hero"
        type="button"
      >
        x
      </button>
    </li>
    <li>
      <a
        href="/detail/8"
      >
        <span
          class="badge"
        >
          8
        </span>
         Nicholas Runolfsdottir V 
      </a>
      <button
        class="delete"
        title="delete hero"
        type="button"
      >
        x
      </button>
    </li>
    <li>
      <a
        href="/detail/9"
      >
        <span
          class="badge"
        >
          9
        </span>
         Glenna Reichert 
      </a>
      <button
        class="delete"
        title="delete hero"
        type="button"
      >
        x
      </button>
    </li>
    <li>
      <a
        href="/detail/10"
      >
        <span
          class="badge"
        >
          10
        </span>
         Clementina DuBuque 
      </a>
      <button
        class="delete"
        title="delete hero"
        type="button"
      >
        x
      </button>
    </li>
  </ul>
</app-heroes>
`;
